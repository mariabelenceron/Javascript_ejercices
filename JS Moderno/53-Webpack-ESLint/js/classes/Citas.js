// CLasses
class Citas {
  constructor() {
    this.citas = [];
  }

  agregarCita(cita) {
    this.citas = [...this.citas, cita];
  }

  editarCita(citaActualizada) {
    // eslint-disable-next-line max-len
    this.citas = this.citas.map((cita) => (cita.id === citaActualizada.id ? citaActualizada : cita));
  }

  eliminarCita(id) {
    this.citas = this.citas.filter((cita) => cita.id !== id);
  }
}

export default Citas;
