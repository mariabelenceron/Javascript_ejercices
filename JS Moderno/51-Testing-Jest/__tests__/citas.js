import Citas from '../js/classes/Citas';

describe('Probar clases de citas', () => {

    const citas = new Citas();
    const id = Date.now();


    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Belen',
            telefono: '0997702037',
            fecha: '10-12-2021',
            hora:'10:30',
            sintomas: 'Solo duerme'
        }

        citas.agregarCita(citaObj);

        expect(citas).toMatchSnapshot();
    });

    test('Actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Nuevo nombre',
            propietario: 'Belen',
            telefono: '0997702037',
            fecha: '10-12-2021',
            hora:'10:30',
            sintomas: 'Solo duerme'
        }

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });

    test('Eliminar citas', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    });

});