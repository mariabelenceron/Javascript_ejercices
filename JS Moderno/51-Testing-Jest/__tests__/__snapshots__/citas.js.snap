// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar clases de citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2021",
      "hora": "10:30",
      "id": 1619466064871,
      "mascota": "Nuevo nombre",
      "propietario": "Belen",
      "sintomas": "Solo duerme",
      "telefono": "0997702037",
    },
  ],
}
`;

exports[`Probar clases de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2021",
      "hora": "10:30",
      "id": 1619466064871,
      "mascota": "Hook",
      "propietario": "Belen",
      "sintomas": "Solo duerme",
      "telefono": "0997702037",
    },
  ],
}
`;

exports[`Probar clases de citas Eliminar citas 1`] = `
Citas {
  "citas": Array [],
}
`;
